Makefile

```
@@ -117,6 +117,35 @@ coverage: ## generate coverprofiles from the unit tests
 		fi; \
 	done )
 
+.PHONY: test-cloud-storage
+test-cloud-storage: start-cloud-storage run-s3-tests stop-cloud-storage ## run cloud storage driver tests
+
+.PHONY: start-cloud-storage
+start-cloud-storage: ## start local cloud storage (minio)
+	docker compose -f tests/docker-compose-storage.yml up minio minio-init -d
+
+.PHONY: stop-cloud-storage
+stop-cloud-storage: ## stop local cloud storage (minio)
+	docker compose -f tests/docker-compose-storage.yml down
+
+.PHONY: reset-cloud-storage
+reset-cloud-storage: ## reset (stop, delete, start) local cloud storage (minio)
+	docker compose -f tests/docker-compose-storage.yml down
+	@mkdir -p tests/miniodata/distribution
+	@rm -rf tests/miniodata/distribution/* tests/miniodata/.minio.sys
+	docker compose -f tests/docker-compose-storage.yml up minio minio-init -d
+
+.PHONY: run-s3-tests
+run-s3-tests: ## run S3 storage driver integration tests
+	AWS_ACCESS_KEY=distribution \
+	AWS_SECRET_KEY=password \
+	AWS_REGION=us-east-1 \
+	S3_BUCKET=images-local \
+	S3_ENCRYPT=false \
+	REGION_ENDPOINT=http://127.0.0.1:9000 \
+	S3_SECURE=false \
```

Forgot one, I also needed this

```suggestion
	S3_SECURE=false \
	S3_ACCELERATE=false \
	AWS_S3_FORCE_PATH_STYLE=true \
```

---

we should also update the tests - IIRC we have some conflicting default S3 driver parameters set in the tests